docs/metrics.md

```diff
@@ -30,8 +126,9 @@ Default controller metrics can be scraped from service ```workflow-controller-me
 
 Metrics that inform on the state of a Workflow, or a series of Workflows. These custom metrics are defined by the user in the Workflow spec.
 
-Emitting custom metrics is the responsibility of the emitter owner. Since the user defines Workflows in Argo, the user is responsible
-for emitting metrics correctly.
+Emitting custom metrics is the responsibility of the emitter owner. Since the user defines Workflows in Argo, the user is responsible for emitting metrics correctly.
+
+Currently custom metrics and their labels must be valid prometheus and OpenTelemetry metric names, which limits them to alpha-numerics, `_`. This applies even if you're only using OpenTelemetry for metrics.
```

Comment:
```suggestion
Currently custom metrics and their labels must be valid Prometheus and OpenTelemetry metric names, which limits them to alphanumeric characters and `_`.
This applies even if you're only using OpenTelemetry for metrics.
```

- use consistent wording for "alphanumeric characters"
- fix missing "and"
- prefer 1 sentence per line of markdown, per style guide

This can also likely be simplified, per k8s style guide